packages/language-server/src/plugins/typescript/CSSClassDefinitionLocator.ts

```diff
@@ -0,0 +1,203 @@
+     */
+    private isStandardClassFormat() {
+        if (this.initialNodeAt?.type == 'Text' && this.initialNodeAt?.parent?.name == 'class') {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * HTML Element target:
+     * <main>
+     */
+    private isTargetHTMLElement() {
+        if (this.initialNodeAt?.type == 'Element') {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Conditional Expression format:
+     * class="{current === 'baz' ? 'selected' : ''
+     */
+    private isConditionalExpressionClassFormat() {
+        if (
+            this.initialNodeAt?.type == 'Literal' &&
+            this.initialNodeAt?.parent?.type == 'ConditionalExpression' &&
+            this.initialNodeAt?.parent.parent?.parent?.name == 'class'
+        ) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Class Directive format:
+     * class:active="{current === 'foo'}"
+     */
+    private isDirectiveFormat() {
+        if (this.initialNodeAt?.type == 'Class' && this.initialNodeAt?.parent?.type == 'Element') {
+            return true;
+        }
+
+        return false;
+    }
+
+    private getStandardFormatClassName() {
+        const testEndTagRange = Range.create(
+            Position.create(this.position.line, 0),
+            Position.create(this.position.line, this.position.character)
+        );
+        const text = this.document.getText(testEndTagRange);
+
+        let loopLength = text.length;
+        let testPosition = this.position.character;
+        let spaceCount = 0;
+
+        //Go backwards until hitting a " and keep track of how many spaces happened along the way
+        while (loopLength) {
+            const testEndTagRange = Range.create(
+                Position.create(this.position.line, testPosition - 1),
+                Position.create(this.position.line, testPosition)
+            );
+            const text = this.document.getText(testEndTagRange);
+            if (text === ' ') {
+                spaceCount++;
+            }
+
+            if (text === '"') {
+                break;
+            }
+
+            testPosition--;
+            loopLength--;
+        }
+
+        const cssClassName = this.initialNodeAt?.data.split(' ')[spaceCount];
+
+        return this.getDefinitionRangeWithinStyleSection(`.${cssClassName}`);
+    }
+
+    private getHTMlElementName() {
+        const result = this.getDefinitionRangeWithinStyleSection(`${this.initialNodeAt.name}`);
+        if (result) {
+            //Shift start/end to get the highlight right
+            const originalStart = result.start.character;
+            result.start.character -= 1;
+            if (this.initialNodeAt.name) {
+                result.end.character = originalStart + this.initialNodeAt.name.length;
+            }
+
+            return result;
+        }
+    }
+
+    private getDefinitionRangeWithinStyleSection(targetClass: string) {
```

Comment:
Apologies, the comment on line `getCSSClassDefinition` was actually meant to appear here. That code does a somewhat unperformant string search. I believe it could be more promising/faster to defer CSS parsing to vscode-css-languageservice which has a parse method. The AST that is returned doesn't have type definitions because it's private, but I'm honestly not worried too much about that since it's unlikely to change for this. That AST could be traversed to more easily find the correct node to link to.